torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""National Agriculture Imagery Program (NAIP) dataset."""
+
+from typing import Any, Dict, Optional
+
+import matplotlib.pyplot as plt
+
+from .geo import RasterDataset
+
+
+class NLCD2016TreeCanopy(RasterDataset):
+    """National Land Cover Database 2016 (NLCD2016) - Tree Canopy dataset.
+
+    The `National Land Cover Database <https://www.mrlc.gov/>`_ provides 30m tree
+    canopy cover raster datasets generated by the United States Forest
+    Service (USFS). The data covers the conterminous United States, coastal Alaska,
+    Hawaii, and Puerto Rico and consists of continuous percent tree canopy estimates
+    for each pixel across all land covers. The data is derived from multi-spectral
+    Landsat imagery and other available ground and ancillary information.
+
+    If you use this dataset in your research, please cite it using the following format:
+
+    * https://doi.org/10.14358/PERS.78.7.715
+
+    .. note::
+
+       This dataset can be automatically downloaded and converted to geotiffs from the
+       ERDAS IMAGINE .img format using the following bash script:
```

Comment:
Any reason to convert to GeoTIFFs? Should also load fine as .img files (although more slowly).

Reply:
I wasn't able to load them without rasterio errors but I'll try again tomorrow

Reply:
More info than you ever wanted to know about img vs. cog below